curriculum/challenges/english/25-front-end-development/quiz-javascript-data-structures/66edd5583761e4f6dae19606.md

```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
+The list is circular by default
+
 
 #### --answer--
 
-Placeholder answer
+Each node contains data and a pointer to the next node
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the primary benefit of using a hash set over an array?
 
 #### --distractors--
 
-Placeholder distractor 1
+Reduced memory usage
 
 ---
 
-Placeholder distractor 2
+Random access to elements
 
 ---
 
-Placeholder distractor 3
+Maintaining order of elements
 
 #### --answer--
 
-Placeholder answer
+Faster insertion and search times due to hash function
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which operation is faster in a doubly linked list compared to a singly linked list?
 
 #### --distractors--
 
-Placeholder distractor 1
+Searching for an element
 
 ---
 
-Placeholder distractor 2
+Reversing the list
 
 ---
 
-Placeholder distractor 3
+Insertion at the end
 
 #### --answer--
 
-Placeholder answer
+Deleting a node when a reference to that node is provided
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the main difference between a queue and a stack?
 
 #### --distractors--
 
-Placeholder distractor 1
+Stack allows random access while queue does not
 
 ---
 
-Placeholder distractor 2
+Queue follows LIFO and stack follows FIFO
 
 ---
 
-Placeholder distractor 3
+Both follow the same order of operations
 
 #### --answer--
 
-Placeholder answer
+Stack follows LIFO and queue follows FIFO
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the time complexity of accessing an element in a static array?
 
 #### --distractors--
 
-Placeholder distractor 1
+O( log n)
 
 ---
 
-Placeholder distractor 2
+O(n log n)
 
 ---
 
-Placeholder distractor 3
+O(n)
 
 #### --answer--
 
-Placeholder answer
+O(1)
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the order of insertion and deletion in a stack?
 
 #### --distractors--
 
-Placeholder distractor 1
+Random access
 
 ---
 
-Placeholder distractor 2
+First in, last out (FILO)
 
 ---
 
-Placeholder distractor 3
+First in, first out (FIFO)
 
 #### --answer--
 
-Placeholder answer
+Last in, first out (LIFO)
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+A set data structure: 
 
 #### --distractors--
 
-Placeholder distractor 1
+Guarantees sorted order
 
 ---
 
-Placeholder distractor 2
+Allows duplicate elements
 
 ---
 
-Placeholder distractor 3
+Allows access by index
 
 #### --answer--
 
-Placeholder answer
+Does not allow duplicate elements
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What differs a dynamic array from a static array?
 
 #### --distractors--
 
-Placeholder distractor 1
+Dynamic arrays require less memory than Static Arrays
 
 ---
 
-Placeholder distractor 2
+Static arrays are always slower
 
 ---
 
-Placeholder distractor 3
+Static arrays are only used in Java
 
 #### --answer--
 
-Placeholder answer
+Dynamic arrays can grow whereas Static arrays have fixed size 
```

Comment:
```suggestion
Dynamic arrays can grow whereas static arrays have fixed size 
```

Reply:
Hey!! A big thanks for giving me guidance and sorry for the mistakes!!